test/integration/developer_has_tags_test.rb

```diff
@@ -0,0 +1,72 @@
+require 'test_helper'
+
+class DeveloperHasTagsTest < ActionDispatch::IntegrationTest
+  test "developers has many tags" do
+    tags = create_list(:tag, 4)
+    dev = create(:developer, tags: tags)
+
+    ApplicationController.any_instance.stubs(:current_developer).returns(dev)
+
+    visit '/developer'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.